test/spanningtrees/boruvka.jl

```diff
@@ -0,0 +1,57 @@
+@testset "Boruvka" begin
+
+    g4 = complete_graph(4)
+
+    distmx = [
+        0  1  5  6
+        1  0  4  10
+        5  4  0  3
+        6  10  3  0
+    ]
+
+    vec_mst = Vector{Edge}([Edge(1, 2), Edge(3, 4), Edge(2, 3)])
+    cost_mst = sum(distmx[src(e),dst(e)] for e in vec_mst)
+    #cost_mst = 8.0
+
+    max_vec_mst = Vector{Edge}([Edge(1, 4), Edge(2, 4), Edge(1, 3)])
+    cost_max_vec_mst = sum(distmx[src(e),dst(e)] for e in max_vec_mst)
+    #cost_max_vec_mst = 21.0
+
+    for g in testgraphs(g4)
+    # Testing Boruvka's algorithm
+        @test @inferred(boruvka_mst(g, distmx)) == (vec_mst,cost_mst)
```

Comment:
I think here you relay on the fact that the edges are returned in a particular order. But does the algorithm really guarantee that?  Otherwise you could compare with `issetequal`. Also, because the type `Edge` is also used for directed edges in directed graphs, `Edge(1,2) != Edge(2, 1)` in general, so maybe you have to consider that here too.

Reply:
> I think here you relay on the fact that the edges are returned in a particular order. But does the algorithm really guarantee that? Otherwise you could compare with `issetequal`. Also, because the type `Edge` is also used for directed edges in directed graphs, `Edge(1,2) != Edge(2, 1)` in general, so maybe you have to consider that here too.

I was thinking about just using the total weight (of the result) to check if it is correct, I just wrote the edge set cause people did it in Kruskal.jl and Prim.jl, but I think you want I can try to use issetequal, what do you think would be better?

Reply:
Using a set is probably better. The reason is that you don't necessarily want to guarantee that the edges will be in a particular order in your code (if you do, and you decide to change things, it becomes a breaking change), and since this is in test code, it doesn't really matter about the waste of converting to set and comparing.